langchain/src/chat_models/base.ts

```diff
@@ -82,8 +84,17 @@ export abstract class BaseChatModel<
 
   lc_namespace = ["langchain", "chat_models", this._llmType()];
 
+  cache?: BaseCache;
+
   constructor(fields: BaseChatModelParams) {
     super(fields);
+    if (typeof fields.cache === "object") {
```

Comment:
Should this setup now move to base language model?

Reply:
Good call, refactored